### PR TITLE
Replace packet-infra Helm chart with native Go templates

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area open-source
/kind cleanup
/platform packet

**What this PR does / why we need it**:
In order to prepare for Helm 3 and to go in line with gardener/gardener#2754, the `packet-infra` Helm chart is eliminated and replaced with native Golang templates.
On the way, I updated the dependencies.

**Special notes for your reviewer**:
See discussion in https://github.com/gardener/gardener/pull/3994#pullrequestreview-652016478 ff.

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Comment by vpnachev** [comment]
Needs `make generate`.

File: go.mod [hunk 1]
```diff
@@ -4,11 +4,12 @@ go 1.16
...
 require (
 	github.com/Masterminds/goutils v1.1.1 // indirect
+	github.com/Masterminds/sprig v2.22.0+incompatible
 	github.com/ahmetb/gen-crd-api-reference-docs v0.2.0
 	github.com/coreos/go-systemd/v22 v22.1.0
 	github.com/frankban/quicktest v1.9.0 // indirect
-	github.com/gardener/etcd-druid v0.3.0
-	github.com/gardener/gardener v1.21.0
+	github.com/gardener/etcd-druid v0.5.0
+	github.com/gardener/gardener v1.23.0
```
Comment (timebertt) [inline go.mod:12]:
```suggestion
	github.com/gardener/gardener v1.23.1
```

in order to get https://github.com/gardener/gardener/pull/4089?

File: pkg/controller/infrastructure/templates/main.tpl.tf [hunk 2]
```diff
@@ -4,9 +4,9 @@ provider "packet" {
 
 // Deploy a new ssh key
 resource "packet_project_ssh_key" "publickey" {
-  name = "{{ required "clusterName is required" .Values.clusterName }}-ssh-publickey"
-  public_key = "{{ required "sshPublicKey is required" .Values.sshPublicKey }}"
-  project_id = "{{ required "packet.projectID is required" .Values.packet.projectID }}"
+  name = "{{ .Values.clusterName }}-ssh-publickey"
+  public_key = "{{ .Values.sshPublicKey }}"
+  project_id = "{{ .Values.packet.projectID }}"
```
Comment (timebertt) [inline pkg/controller/infrastructure/templates/main.tpl.tf:9]:
```suggestion
  name = "{{ .clusterName }}-ssh-publickey"
  public_key = "{{ .sshPublicKey }}"
  project_id = "{{ .packet.projectID }}"
```

Right? Same would apply in line 16.

**Review by timebertt — CHANGES_REQUESTED** [review]

File: pkg/controller/infrastructure/templates/main.tpl.tf (hunk 2 above)
Comment (rfranzke) [reply pkg/controller/infrastructure/templates/main.tpl.tf]: Nice catch... :) thanks!

**Review by timebertt — APPROVED** [review]
Thanks!
/lgtm
apart from
```
> Generate / Vendor Check
>> make generate
make generate needs to be run:
 M example/controller-registration.yaml
make: *** [Makefile:97: check-generate] Error 1
```

**Comment by rfranzke** [comment]
Strange... I explicitly regenerated and my diff showed changes in `example/controller-registration.yaml`...

**Comment by rfranzke** [comment]
OK, it seems the issue was helm v2 vs. v3... :)

**Comment by timebertt** [comment]
> OK, it seems the issue was helm v2 vs. v3... :)

Yeah. That's what we're trying to get rid of 😄

**Review by timebertt — APPROVED** [review]
/lgtm

**Pull request merged**